src/rule_with_actions.cc

```diff
@@ -121,10 +122,8 @@ RuleWithActions::RuleWithActions(
                     m_actionsRuntimePos.push_back(a);
                 }
             } else {
+                assert(false && "The handling of RunTimeBeforeMatchAttemptKind has not been implemented yet.");
                 delete a;
-                std::cout << "General failure, action: " << a->m_name;
-                std::cout << " has an unknown type." << std::endl;
-                throw;
```

Comment:
What we are doing here is a switch over `a->action_kind` withe the 2 cases `Action::ConfigurationKind` and `Action::RunTimeOnlyIfMatchKind`.

This `else` would be the default case of the switch, right.

Would it be better to replace the if/else with a switch and ask the compiler of our switch case is complete (e.g. all cases are implemented)

----

independent from this remark: Is the else case a "can never happen" or is it a "user configured invalid modsec code". In the second case, we change the semantic which could be a problem.

Reply:
The refactoring of the if-then chain to a switch statement is complete. We reviewed the code together with @airween  and couldn't find a way to force execution into the else/default case.